### ISSUE TITLE
Add option to disable default metric paths in ratpack-codahale-metrics

### ISSUE THREAD

**Issue by drmaas** [ratpack/ratpack#756]
When this module is installed, by default all unique paths result in a new metric. For large data sets the number of metrics becomes unbounded. If we're sending metrics to external systems like statsd, the network traffic becomes unmanageable. 

I know there is a way of collapsing metrics by group name, but the default metric handler/interceptor still execute. It would be useful to provide an option to disable the default metric handler/interceptor so that we could drop in our own handler/interceptor, while taking advantage of the other features provided by the module.

**Comment by alkemist** [comment]
@rhart seems reasonable to me. WDYT?

**Comment by rhart** [comment]
Sure, makes sense.  I'd still like to understand this a little deeper though.  To see whether there is anything else we can be doing.

@drmaas are you saying that the metric collapsing solves your network traffic issues but you don't want every request to go through the handler?  Even if we allowed requests to be excluded at the handler/interceptor level all requests would still go through the handler, just not all recorded.  But would adding this functionality help?  What would your handler/interceptor do differnet?

**Comment by drmaas** [comment]
It looks like request and execution metrics are recorded. We are only interested in recording request metrics, and we'd like to record timers for the following names:

1) api.response-code.{response code}
2) api.{method}.response-code.{response code}
3) all

It doesn't look like we can achieve that with the current config. If that is indeed the case, we'd like to be able to exclude the injection of ```CodaHaleMetricsModule.HandlerDecoratorProvider```, and instead inject our own handler.

**Comment by rhart** [comment]
Ok, I see.  Will defo put a config option in so you can switch this off.  

Wondering whether we can also allow you to set your own strategy for defining how the metrics are captured, with what we have now as the default.  That way you get the handler decorating and timer capture for free and you can easily provide your own metric format.  Will look into this too.

**Comment by drmaas** [comment]
I think it would be cool to have a config option to allow us to override the default handler with our own - it seems like an elegant solution. It would still be useful to be able to disable it completely for those cases where the user doesn't want to collect any request metrics.

**Comment by drmaas** [comment]
It looks like this is still hardcoded - ideally we could turn off the default handler and blocking interceptor. Let me know if I'm missing something.

```
    bind(RequestTimingHandler.class);
    bind(BlockingExecTimingInterceptor.class);

    Provider<RequestTimingHandler> handlerProvider = getProvider(RequestTimingHandler.class);
    Multibinder.newSetBinder(binder(), HandlerDecorator.class).addBinding().toProvider(() -> HandlerDecorator.prepend(handlerProvider.get()));
 ```

**Comment by rhart** [comment]
Yeah, this issue didn't make the cut for 1.0.

I will try and get done for 1.2 but if you can submit a PR that would be awesome

**Comment by drmaas** [comment]
I've got something quasi-working locally - with the ability to disable the request timing handler and blocking exec timing interceptor, use the defaults, or provide custom implementations. Will do a PR when I get the tests completed.

**Comment by rhart** [comment]
Awesome

**Comment by rhart** [comment]
Fixed by #832
